src/asn1.mli

```diff
@@ -238,6 +278,14 @@ sig
   val dh_grammar : DH.Params.t Asn.t
 end
 
+module CVC :
```

Comment:
I do not think that anything will need to call a `CVC.decode` with this signature. So you can create functions with signature `decode : Cstruct.t -> (RSA_CVC.Public.t, string) Result.result` (resp. `ECDSA_CVC.Public.t`) and only expose those.

- In a first step, those can be implemented in terms of the current `decode` function
- In a second step, that should be an opportunity to simplify your code since you do not need to perform detection (that can be done in a later PR).